src/ansys/mapdl/core/reader/mesh.py

```diff
@@ -0,0 +1,391 @@
+        return self._secnum
+
+    @property
+    def element_coord_system(self):
+        """Element coordinate system number"""
+        if self._esys is None:
+            self._esys = self._elem[self._elem_off[:-1] + 4]  # FIELD 4
+        return self._esys
+
+    @property
+    def elem(self):
+        """List of elements containing raw ansys information.
+
+        Each element contains 10 items plus the nodes belonging to the
+        element.  The first 10 items are:
+
+        - FIELD 0 : material reference number
+        - FIELD 1 : element type number
+        - FIELD 2 : real constant reference number
+        - FIELD 3 : section number
+        - FIELD 4 : element coordinate system
+        - FIELD 5 : death flag (0 - alive, 1 - dead)
+        - FIELD 6 : solid model reference
+        - FIELD 7 : coded shape key
+        - FIELD 8 : element number
+        - FIELD 9 : base element number (applicable to reinforcing elements only)
+        - FIELDS 10 - 30 : The nodes belonging to the element in ANSYS numbering.
+
+        """
+        if self._cached_elements is None:
+            self._cached_elements = np.split(self._elem, self._elem_off[1:-1])
+        return self._cached_elements
+
+    @property
+    def enum(self):
+        """ANSYS element numbers."""
+        if self._enum is None:
+            self._enum = self._elem[self._elem_off[:-1] + 8]
+        return self._enum
+
+    @property
+    def nnum(self):
+        """Array of node numbers."""
+        return self._nnum
+
+    @property
+    def ekey(self):
+        """Element type key
+
+        Array containing element type numbers in the first column and
+        the element types (like SURF154) in the second column.
+
+        """
+        return self._ekey
+
+    @property
+    def rlblock(self):
+        """Real constant data from the RLBLOCK."""
+        # if not self._rdat:
+        #     pass # todo: fix this
+        # return self._rdat
+        raise NotImplementedError()
+
+    @property
+    def rlblock_num(self):
+        """Indices from the real constant data"""
+        # if not self._rnum:
+        #     pass # todo: to fix
+        # return self._rnum
+        raise NotImplementedError()
+
+    @property
+    def nodes(self):
+        """Array of nodes."""
+        if self._node_coord is None:
+            self._node_coord = np.ascontiguousarray(self._nodes[:, :3])
+        return self._node_coord
+
+    @property
+    def node_angles(self):
+        """Node angles from the archive file."""
+        # if self._node_angles is None:
+        #     self._node_angles = np.ascontiguousarray(self._nodes[:, 3:])
+        # return self._node_angles
+        raise NotImplementedError()
+
+    def _parse_vtk(
+        self,
+        allowable_types=None,
+        force_linear=False,
+        null_unallowed=False,
+        fix_midside=True,
+        additional_checking=False,
+    ):
+        from ansys.mapdl.reader.mesh import Mesh
```

Comment:
@germa89 what is going on here?

Reply:
As we said, we will keep the legacy dependency for the ``_parse_vtk`` until we sort out the extra library or operator.

Reply:
@germa89 Passing "self" into a member function on a class that is not a base class is not correct